baseapp/internal/protocompat/protocompat.go

```diff
@@ -206,3 +206,17 @@ func isProtov2(md grpc.MethodDesc) (isV2Type bool, err error) {
 	_, _ = md.Handler(nil, nil, pullRequestType, doNotExecute)
 	return
 }
+
+// RequestFullNameFromMethodDesc returns the fully-qualified name of the request message of the provided service's method.
+func RequestFullNameFromMethodDesc(sd *grpc.ServiceDesc, method grpc.MethodDesc) (protoreflect.FullName, error) {
+	methodFullName := protoreflect.FullName(fmt.Sprintf("%s.%s", sd.ServiceName, method.MethodName))
```

Comment:
does this assume protoreflect?

Reply:
it multiplexes between protov2 API and gogoAPI; but besides that it's hybrid and protoreflect provides a nice way to parse the full protobuf name stack.